pkg/odo/commonflags/variables.go

```diff
@@ -0,0 +1,57 @@
+package commonflags
+
+import (
+	"errors"
+	"os"
+
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/testingutil/filesystem"
+	"github.com/redhat-developer/odo/pkg/vars"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	VarFileFlagName = "var-file"
+	VarFlagName     = "var"
+)
+
+// UseVariablesFlags indicates that a command accepts the --var-file and --var flags
+func UseVariablesFlags(cmd *cobra.Command) {
+	if cmd.Annotations == nil {
+		cmd.Annotations = map[string]string{}
+	}
+	cmd.Annotations["variables"] = "true"
+}
+
+// AddVariablesFlags adds the --var-file and --var flags to all commands
+// We use "flag" in order to make this accessible throughtout ALL of odo, rather than the
+// above traditional "persistentflags" usage that does not make it a pointer within the 'pflag'
+// package
+func AddVariablesFlags() {
+	pflag.CommandLine.StringArray(VarFlagName, []string{}, "Variable to override Devfile variable and variables in var-file")
+	pflag.CommandLine.String(VarFileFlagName, "", "File containing variables to override Devfile variables")
```

Comment:
Just a discussion (also applies to the `run-on` flag):

From the user perspective, now that these two flags are visible globally, can the `odo help` output provide more guidance by telling users which commands these flags can be used with? I understand this would mean maintaining these descriptions, but I think it would be helpful.
So I'd suggest something like:

```suggestion
	pflag.CommandLine.String(VarFileFlagName, "", "File containing variables to override Devfile variables. Usable with \"odo dev\" and \"odo deploy\" commands.")
```

What do you think?

Because currently, users have to explicitly run the command to see if the flag can be used or not, e.g.:

```
> odo init --var USER                                                             
 ✗  --var-file and --var flags are not supported for this command
```

Reply:
I would like to try and do it in an automatic way, to save maintenance

Reply:
Yup, it would be even better if we could do it automatically 👍🏿 
If it is not too much work, we can do it in a separate issue/PR though.

Reply:
ok, let's work on this in another PR